### PR TITLE
Updated bench scripts for strings

### PR DESCRIPTION
**Pull request by R4PaSs**
Updated the script for benching the variants of strings.

Some new variations for strings will follow in further PRs

### CODE REVIEW

**Comment by Morriar** [comment]
You forget to commit the README file that explain the objective of each bench :)

**Comment by xymus** [comment]
So much mysterious code that could benifit from comments... but I'm gonna assume that there will soon be a document to explain them? Let's say 70 to 100 pages long?

so :shipit:

**Comment by privat** [comment]
Pictures with colors and bars are missing in this PR

**Comment by privat** [comment]
chez moi, ça ça crashes (segfault):
~~~sh
$ string_concat/chain_concat -m ropebuf --ccts 1 --loops 1 --strlen 1
~~~

File: benchmarks/strings/bench_strings.sh [hunk 1]
```diff
@@ -40,21 +38,20 @@ function usage()
...
+		mkdir arraytos
+	fi
+	cd arraytos
+
 	if $verbose; then
 		echo "*** Benching Array.to_s performance ***"
 	fi
 
-	../bin/nitc --global ./strings/array_tos.nit -m ./strings/array_to_s_vars/array_to_s_flatstr.nit -m ../lib/standard/ropes.nit --make-flags "CFLAGS=\"-g -O2 -DNOBOEHM\""
+	../../../bin/nitc --global ../array_tos.nit -m ../array_to_s_vars/array_to_s_rope.nit --make-flags "CFLAGS=\"-g -O2 \""
```
Comment (privat) [inline benchmarks/strings/bench_strings.sh]: tu peux virer les --make-flags car sans le -DNOBOEHM c'est la config par defaut

**Comment by privat** [comment]
La plupart des trucs su .sh pourraient se factoriser en des fonctions qui prennent que ce qui change et qui s'occuperait de la compil, des executions, des affichages.

**Comment by privat** [comment]
A first graph for concat (note that bufrope chrashes  so no numbers)

**Comment by privat** [comment]
Other diagrams

substring:

substr

iteration (unreadable):

iter

arrat_to_s:

array_tos


because the x axis is numbers, maybe draw lines instead of bars, they could make the diagram more easier to read

**Comment by R4PaSs** [comment]
OK @Morriar, there's a README for you now !

@xymus, coming ~~soon~~

And @privat I removed the flags, and the RopeBuffer stuff should be corrected now !

graph

**Comment by privat** [comment]
@Morriar dois-je attendre un meilleur README?

**Comment by Morriar** [comment]
@privat you are the boss, you decide

**Pull request merged**